Truncate completion options which are too long

airblade opened the pull request:
Fixes #226.

adamsmith commented:
What do you think about making it 65? Were there cases where that seemed too low?

airblade commented:
I made the width a bit larger because when the `display` value is wide, there isn't much room left for the `hint`.

Here's a screenshot with a max width of 65:

<img width="466" alt="Screenshot 2019-10-17 at 6 13 14 pm" src="https://user-images.githubusercontent.com/7265/67031636-db6f1280-f109-11e9-9066-ec649ab2d8f5.png">

(Please ignore the misalignment on the right hand side for now; that's a bug which wasn't there when the width was 75, and which I'll fix.)

airblade commented:
Alternatively we could calculate the available width of the window and use that (with some padding) as a maximum width.

airblade commented:
I have fixed the alignment bug above.

I also extracted `g:kite_completion_max_width` (default `75`) so the user can tweak it easily.

@adamsmith The width is a trade-off between those occasions when the LHS is large (e.g. `json.dump()` call signatures), and those when the RHS is large (e.g. union types).  I have been assuming that we should never truncate the LHS.

adamsmith commented:
Ok sounds good to me @ 75.

@its-dhung please weigh in if you have any opinions or thoughts.

Thanks all!

airblade commented:
@jansorg The completion menu takes only as much space as needed now.

It sets its max width to the smaller of `g:kite_completion_max_width` (default `75`) and the remaining space in the window.  If the actual width would be wider than the max width, it truncates first the right hand side (while keeping the Kite symbol) and then the left hand side.

jansorg commented:
thank you @airblade , this is much better.
Is it possible that some extra padding is applied when computing the max width? In a small window it's not spanning all remaining space on the right. Would it be possible to use more of the available space in a case like this?

(Linux, gVim)
![Screenshot_20191030_113957](https://user-images.githubusercontent.com/11473063/67851163-1e69b680-fb0a-11e9-8364-39f8fa1f3175.png)

airblade commented:
@jansorg It was underestimating the width in the case where there was no sign column and/or no number column.  It should be more accurate now.
